pkg/compact/compact.go

```diff
@@ -724,40 +724,49 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 
 	// Once we have a plan we need to download the actual data.
 	begin := time.Now()
+	toCompactDirs := make([]string, len(toCompact))
 
-	toCompactDirs := make([]string, 0, len(toCompact))
-	for _, meta := range toCompact {
+	var eg errgroup.Group
+	for i, meta := range toCompact {
+		meta := meta
 		bdir := filepath.Join(dir, meta.ULID.String())
+		toCompactDirs[i] = bdir
```

Comment:
This line of code does not run concurrently, so we can turn this back into the more idiomatic style we tend to use
```suggestion
		toCompactDirs = append(toCompactDirs, bdir)
```